### PR TITLE
Force to execute testcases sequentially in a test suite

### PR DESCRIPTION
**Pull request by raoyitao**
* An attribute "strict_order" is added into test suite and testcases
  in this suite must run sequentially one by one and parallel feature
  such as execution group is disabled in batch mode.
* A test suite with "strict_order" ignores `test_sorter` configured for
  Test runner, and `test_filter` config does not take effect unless all
  of the testcases in suite been filtered out.
* If "strict_order" is specified, on interactive UI a testcase entry is
  disabled unless previous ones in the same test suite have run, this
  also applies to testcases in a parametrization group. At server side
  disorded execution can be detected and an error message is returned.
* In interactive mode the report of a multitest will be reset after
  environment is restarted.
* Fix an issue that `ProcessRunnerTest` & `PyTest` & `PyUnit` have no
  "reset env" button on interactive UI.
* Fix an issue that when "reset env" button is inactive, click event
  should be ignored and no action taken.
* Fix an issue that description of Testplan is missing.
* Fix an potential bug in interactive testcase (test_api).

## Checklist:
- [x] Test
- [ ] Example (both test_plan.py and .rst)
- [x] Documentation (API)
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

**Review by Pyifan — CHANGES_REQUESTED** [review]

File: testplan/runnable/interactive/base.py [hunk 1]
```diff
@@ -345,6 +351,8 @@ def start_test_resources(self, test_uid, await_results=True):
 
         test = self.test(test_uid)
         test.start_test_resources()
+        # Re-initialize report of Multitest if environment restarted
+        self.report[test_uid] = test.dry_run().report
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/runnable/interactive/base.py:355]: for now I cannot know it.

File: testplan/testing/base.py [hunk 2]
```diff
@@ -154,6 +153,7 @@ def _new_test_report(self):
             uid=self.uid(),
             category=self.__class__.__name__.lower(),
             tags=self.cfg.tags,
+            env_status=ResourceStatus.STOPPED,
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/testing/base.py:155]: Actually, this init value is coded in the base class ``Test``. Seems we don't need to know it the env starts at 1st time or restart.

File: testplan/testing/multitest/base.py [hunk 3]
```diff
@@ -276,25 +276,43 @@ def get_test_context(self, test_filter=None):
...
+                testcases
+                if getattr(suite, "strict_order", False)
+                else self.cfg.test_sorter.sorted_testcases(testcases)
             )
 
             testcases_to_run = [
                 case
                 for case in sorted_testcases
-                if test_filter.filter(test=self, suite=suite, case=case)
+                if (test_filter or self.cfg.test_filter).filter(
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/testing/multitest/base.py:301]: I removed a line in this function.

``test_filter`` is passed when interactive handler filters testcases, while ``self.cfg.test_filter`` is used as normal

File: testplan/runnable/interactive/base.py [hunk 4]
```diff
@@ -782,13 +802,33 @@ def _merge_testcase_reports(self, testcase_reports):
...
             with self.report_mutex:
-                parent_entry = self.report
+                disordered = False
+                parent_entry, entry = self.report, None
                 for uid in parent_uids:
-                    parent_entry = parent_entry[uid]
+                    entry = parent_entry[uid]
+                    if getattr(
+                        parent_entry, "strict_order", False
+                    ) and not self._prev_testcases_finished(
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/runnable/interactive/base.py]: Yes, need refactor.
This check can happen after ``_should_run`` check, then directly return an error message to client, but does not make changes to the current report.

**Review by Pyifan — DISMISSED** [review]

**Review by Pyifan — APPROVED** [review]

**Pull request merged**